allennlp/training/optimizers.py

```diff
@@ -196,6 +205,159 @@ def default(model_parameters: List) -> "Optimizer":
         return Optimizer.from_params(model_parameters=model_parameters, params=Params({}))
 
 
+@Optimizer.register("regex")
+class RegexOptimizer(Optimizer):
+    """
+    A `RegexOptimizer` creates a dictionary of `Optimizer`s keyed on some 'name'.
+    Each Optimizer contains its own set of parameters which are obtained using
+    regex matches for certain model parameters.
+    
+    This optimizer works by taking in a parameter `optimizers` which contains a list of `Optimizers`
+    with their keyword arguments, and a parameter `parameter_groups`, which contains regexes and their
+    corresponding optimizer and optional non-default optimizer options for this group. 
+    The regexes in the parameter groups are assigned to their optimizer based on the 'name' argument
+    where the 'name' value should be the same for the optimizer and parameter group.
+    You should specify a default optimizer with 'name': 'default' which will be used for all 
+    parameters which didn't obtain a regex match or when your parameter group doesn't contain a 'name' 
+    parameter.
+
+    # Parameters
+
+    optimizers: `List[Dict[str, Any]]`
+        A list of optimizers to use. Each entry in the list is a dictionary of keyword arguments. A 'name' 
+        keyword argument should be given which will serve as the key to match the optimizer with a 
+        specific parameter group. You should also supply an entry for the default parameter group, 
+        e.g. 'name': 'default'.
+
+    parameter_groups:  `List[Tuple[List[str], Dict[str, Any]]`, optional (default = `None`)
```

Comment:
I had originally allowed this parameter to be `None` where everything would be appended to the `default` group but really you only want to use this class if you have different parameter groups and it required more code to handle so it is now a required parameter.

Reply:
It still says `optional` though?

Reply:
Yes my bad, I only noticed I kept that there when reviewing.